docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,276 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream utilities for downloading files, it is time to introduce you to additional *upstream* utilities. These are utilities that publish files to the Hub from your library. This guide will show you how to:
+
+* Use the `HfApi` class to manage a repository.
+* Use the `Repository` class to handle files and version control a repository with Git-like commands.
+
+## `HfApi`
+
+The `HfApi` class is a high-level class that wraps around HTTP requests. There are many valuable tasks you can accomplish with the `HfApi` class, including: 
+
+- List and filter models and datasets.
+- Inspect model or dataset metadata.
+- Delete a repository.
+- Change the visibility of a repository.
+
+### List and filter
+
+It can be helpful for users to see a list of available models and filter them according to a specific language or framework. This can be especially useful for library and organization owners who want to view all their models. Use the `list_models` function with the `filter` parameter to search for specific models.
+
+You can view all the available filters on the Hugging Face Hub website.
+
+![/docs/assets/hub/hub_filters.png](/docs/assets/hub/hub_filters.png)
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List all models.
+>>> api.list_models()
+
+# List only text classification models.
+>>> api.list_models(filter="text-classification")
+
+# List only Russian models compatible with PyTorch.
+>>> api.list_models(filter=("ru", "pytorch"))
+
+# List only the models trained on the "common_voice" dataset.
+>>> api.list_models(filter="dataset:common_voice")
+
+# List only the models from the spaCy library.
+>>> api.list_models(filter="spacy")
+```
+
+Explore available public datasets with `list_datasets`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List only text classification datasets.
```

Comment:
Do we have documentation somewhere for `task_categories` and `task_ids`?

Reply:
I was only able to find dataset `task_categories` [here](https://github.com/huggingface/huggingface_hub/blob/379414dade73e0816a996d83ead7e4ffd6986a53/docs/hub/datasets-tags.md), but the links are broken. Couldn't really find anything for `task_ids` except for the example [here](https://github.com/huggingface/huggingface_hub/blob/8e2b37c830d43ad3d81b866f2dc3c7778a6da358/src/huggingface_hub/hf_api.py#L463).